### PR TITLE
The first quantum networks may not be "routing and forwarding" networks.

### PR DESCRIPTION
**Pull request by ngym**
Bell pairs are undirected network resources. This fact causes many differences between quantum network architectures and classical network architectures. It should be worth clarifying.

### CODE REVIEW

File: draft-irtf-qirg-principles-03.xml
Comment (Wojtek242) [inline draft-irtf-qirg-principles-03.xml:678]: I'm not sure what is actually meant by the paragraph below. What is residual entanglement? Do you mean that whilst photons are transmitted in a directional manner, the resulting entangled pair is not directed?

File: draft-irtf-qirg-principles-03.xml
Comment (Wojtek242) [inline draft-irtf-qirg-principles-03.xml:821]: This title is confusing. Maybe something simpler? E.g. "First generations of quantum networks will not be store-and-forward" perhaps?

File: draft-irtf-qirg-principles-03.xml
Comment (Wojtek242) [inline draft-irtf-qirg-principles-03.xml:825]: This paragraph needs a bit more detail on what you mean by this difference.

File: draft-irtf-qirg-principles-03.xml
Comment (Wojtek242) [inline draft-irtf-qirg-principles-03.xml:835]: If routing is the same, why not leave it out? Only contrast the data plane activity. Otherwise, I was reading expecting some explanation of how routing will be different, but this is not the point of this paragraph.

File: draft-irtf-qirg-principles-03.xml
Comment (Wojtek242) [inline draft-irtf-qirg-principles-03.xml]: The first sentence is too detailed in technical details. Since entanglement swapping is already explained in this document, why not refer to it?

File: draft-irtf-qirg-principles-03.xml
Comment (Wojtek242) [inline draft-irtf-qirg-principles-03.xml]: Is "routing and measuring" that important to mention? It feels like a side-note and I think it's best to focus on the core ideas in this document

File: draft-irtf-qirg-principles-03.xml
Comment (Wojtek242) [inline draft-irtf-qirg-principles-03.xml:846]: Either this paragraph needs to be much clearer, or, if we add a section on the three generations, this can go there.

**Review by Wojtek242 — CHANGES_REQUESTED** [review]
First, thank you for your contribution!

Overall, I think your contribution makes a very important point that will be very useful for people coming from a classical background. My comments are mostly about clarifying what you meant. 

I am also not sure about this "routing and forwarding" and "routing and swapping" terminology. Is it common? I know of "store-and-forward" and perhaps we can say "store-and-swap" which highlights the need for memory (some proposals do not need a memory, but most practical implementations so far rely on memory as far as I know). Also "store-and-swap" suggests we can keep things in memory for a long time which is not the case. I think this would be a good point to discuss. I would definitely suggest leaving routing completely out of this section as it is not actually discussed at all.

**Comment by Wojtek242** [comment]
I think I messed up the comments - they're meant to refer to the text below, not above - hope it's clear from the context though.

File: draft-irtf-qirg-principles-03.xml
Comment (ngym) [reply draft-irtf-qirg-principles-03.xml:678]: You are correct. I mean that while photons are transmitted in a directional manner, the resulting entangled pair is not directed.

File: draft-irtf-qirg-principles-03.xml
Comment (ngym) [reply draft-irtf-qirg-principles-03.xml:821]: "First generations of quantum networks will not be store-and-forward" is fine.

File: draft-irtf-qirg-principles-03.xml
Comment (ngym) [reply draft-irtf-qirg-principles-03.xml]: "Routing and measuring" includes more generations. Hence,  routing and measuring can be omitted to focus on the first generations. Adding a section to explain the three generations will work well.

File: draft-irtf-qirg-principles-03.xml
Comment (ngym) [reply draft-irtf-qirg-principles-03.xml:846]: Yep, I'll think about adding the section.

File: draft-irtf-qirg-principles-03.xml
Comment (ngym) [reply draft-irtf-qirg-principles-03.xml]: I agree.

File: draft-irtf-qirg-principles-03.xml
Comment (ngym) [reply draft-irtf-qirg-principles-03.xml]: Finally, I removed discussion related to measuring.

File: draft-irtf-qirg-principles-03.xml
Comment (Wojtek242) [inline draft-irtf-qirg-principles-03.xml]: I don't think this is your e-mail address :)

File: draft-irtf-qirg-principles-03.xml
Comment (Wojtek242) [inline draft-irtf-qirg-principles-03.xml]: Hmm, this still isn't super clear to me. How about:
```suggestion
         <t> Note that whilst photons are emitted in a particular direction the 
         resulting entangled pair of qubits does not have a direction associated 
         with it. Physically, there is no upstream or downstream end of the 
         pair.</t>   
```
The second sentence in your original paragraph, whilst correct, I could not see how it supported the first sentence. If this connection was important, could you explain it?

File: draft-irtf-qirg-principles-03.xml
Comment (Wojtek242) [inline draft-irtf-qirg-principles-03.xml]:
```suggestion
           resources, in contrast to directed packets of classical networks.
```

Comment (Wojtek242) [reply draft-irtf-qirg-principles-03.xml]: links don't have to be directional in classical networks either - it is packets/frames/datagrams that have a source and destination

File: draft-irtf-qirg-principles-03.xml
Comment (Wojtek242) [inline draft-irtf-qirg-principles-03.xml]:
```suggestion
           Classical networks receive data on one interface, store it on local buffers, 
           then forward the data to another appropriate interface.
```

File: draft-irtf-qirg-principles-03.xml
Comment (Wojtek242) [inline draft-irtf-qirg-principles-03.xml]:
```suggestion
           Quantum networks store Bell pairs and then
```
I'd say by now the reader should know we mean a single node stores only one qubit of each pair (it's in the entanglement swapping section)

File: draft-irtf-qirg-principles-03.xml
Comment (Wojtek242) [inline draft-irtf-qirg-principles-03.xml]:
```suggestion
           execute entanglement swapping instead of forwarding in the data plane.
```

File: draft-irtf-qirg-principles-03.xml
Comment (Wojtek242) [inline draft-irtf-qirg-principles-03.xml]:
```suggestion
           the order in which the Bell pairs were generated since they are
           undirected. This distinction makes control algorithms and
```

File: draft-irtf-qirg-principles-03.xml
Comment (Wojtek242) [inline draft-irtf-qirg-principles-03.xml]: I don't think this citation is correct.

File: draft-irtf-qirg-principles-03.xml
Comment (Wojtek242) [inline draft-irtf-qirg-principles-03.xml]:
```suggestion
           It is worth noting, that third generation <refer to generations section> 
           quantum repeaters will be able to transmit qubits directly
           in a "store and forward" manner since they will be able to carry
           arbitrary states hop by hop. Thus, the networking algorithms of the
           later generations of quantum networks may end up having more in 
           common with classical networks.</t>
           
```

**Review by Wojtek242 — COMMENTED** [review]
Thanks for implementing the changes. They look good to me!

I would like to spend some more time on the generation section though. I need to think more about what this section should contribute to the document. There is nothing wrong with what you wrote, I'd just rather not include it until I'm clear on what its contents should be and then I'll come back to your contribution.

However, the text about directionality, once you have a look at my proposed changes, can already be included. Since the generation section is currently in the same PR, if you would like your contributions merged sooner, please submit the section about generations as a separate PR, and I will merge the rest.

File: draft-irtf-qirg-principles-03.xml
Comment (ngym) [reply draft-irtf-qirg-principles-03.xml]:
```suggestion
           resources, in contrast to directed frames of classical networks.
```
I think this should be frames because this sentence talk about links.

File: draft-irtf-qirg-principles-03.xml
Comment (ngym) [reply draft-irtf-qirg-principles-03.xml:943]:
```suggestion
           the order in which the Bell pairs are generated since they are
           undirected. This distinction makes control algorithms and
```

File: draft-irtf-qirg-principles-03.xml
Comment (ngym) [reply draft-irtf-qirg-principles-03.xml]: Ooops, it should be sci. repo. in 2016.

File: draft-irtf-qirg-principles-03.xml
Comment (Wojtek242) [inline draft-irtf-qirg-principles-03.xml]: So I've been thinking about this section. Currently it suggests that 3G will likely do store and forward. I think this needs to be relaxed a bit. Or at least made consistent with what I will soon propose with regards to the three generations for the draft. Can we move this to the other pull request too? Rather than merging it now and then changing it. 

Let me know, if you're okay with this and I'll just remove this myself. You'll need to wait for me to merge this request first I guess before you update the other PR.

Comment (ngym) [reply draft-irtf-qirg-principles-03.xml]: No problem. You can remove those sentences by yourself. Please go ahead :)
I'll update the other PR after you merge this request.

Comment (Wojtek242) [reply draft-irtf-qirg-principles-03.xml]:
```suggestion
```

**Pull request merged**
